### ISSUE TITLE
jQuery calls take too many lines

### ISSUE THREAD

**Issue by tqwhite** [jlongster/prettier#521]
This:

`$('body').on`

Becomes this:
```

$(
'body'
).on

```
It's hard to read.

**Comment by tqwhite** [comment]
Until it gets fixed...

I run prettier as a text filter in bbedit. It's a small nodejs program that grabs standard input, runs prettier on standard input and returns it to standard output.

If you add this after you run prettier, it will fix the jquery formatting:

`outString=outString.replace(/\$\(\s*\n\s*(\S+)\n\s*\)/g, '$'+"($1)");`

FYI

**Comment by vjeux** [comment]
Thanks for the report! #469 is a big overhaul of the way those calls are handled. Would you mind proving some concrete examples of code (3-6 lines) so I can make sure they print in a non crazy way!

Thanks

**Comment by vjeux** [comment]
More context would be useful because the example you just put works well. It probably has to do with some other things interacting to each others in bad ways.

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22%24(%5Cn'body'%5Cn).on%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%7D%7D

**Comment by vjeux** [comment]
I'm going to close this as there isn't enough information to go along with. I'll reopen if you provide more info :)

**Comment by tqwhite** [comment]
Honestly, I figured this was straightforward bug. I installed prettier in my live, tried it, had this hassle and pasted it in. At your request, I extracted the lines surrounding the example and, when I tried them on their own, the behavior did not occur. !!!???

Further investigation shows that the problem is actually quite complicated. I started trying to write it all down but, that got really hard. Here's a quick video of the problems (there are actually three screwy behaviors I found).

[http://media.genericwhite.com/video/?prettierIssue013117](http://media.genericwhite.com/video/?prettierIssue013117)

The code is in the file attached. Thanks for the excellent software. Let me know if there is anything I can do to help.


[prettierIssueTqWhiteIi.js.zip](https://github.com/jlongster/prettier/files/743324/prettierIssueTqWhiteIi.js.zip)

**Comment by vjeux** [comment]
Thanks! That's super helpful!

I just landed a change ( #469 ) and your code now looks like this:

```js
self.setRenderEventListeners = function(chartManagementObjects) {
  for (var containerId in chartManagementObjects) {
    $(document).on(
      "hide:" + chartSpecElement.chartId,
      { chartObject: chartManagementObjects[containerId] },
      function() {}
    );

    if (
      !qtools.getSurePath(
        chartSpecElement,
        "enlargedVersionSpec.disallow",
        false
      )
    ) {
      $(chartSpecElement.chartPanelContainer).on(
        "click",
        { chartObject: targetChart },
        function(ev) {}
      );
    }
  } //end of for
}; //end of inner function
```

Could you tell me if it looks good to you? If yes, I'm going to do a release tomorrow and so you can use it without doing a special build.

**Comment by tqwhite** [comment]
This looks very good to me. I am delighted. Thank you very much for the effort. It's a generous thing to do and I admire you for it.
